third-party/ios/ExecutorchLib/ExecutorchLib/runner/runner.cpp

```diff
@@ -26,14 +23,24 @@ using ::executorch::runtime::Result;
 
 namespace llm = ::executorch::extension::llm;
 
+std::string loadBytesFromFile(const std::string &path) {
+  std::ifstream fs(path, std::ios::in | std::ios::binary);
+  if (fs.fail()) {
+    throw std::runtime_error("Failed to open tokenizer file");
+  }
+  std::string data;
+  fs.seekg(0, std::ios::end);
+  size_t size = static_cast<size_t>(fs.tellg());
+  fs.seekg(0, std::ios::beg);
+  data.resize(size);
+  fs.read(data.data(), size);
+  return data;
+}
+
 namespace {
-static constexpr auto kAppendEosToPrompt = "append_eos_to_prompt";
 static constexpr auto kEnableDynamicShape = "enable_dynamic_shape";
-static constexpr auto kBosId = "get_bos_id";
 static constexpr auto kEosIds = "get_eos_ids";
 static constexpr auto kMaxSeqLen = "get_max_seq_len";
-static constexpr auto kNBos = "get_n_bos";
-static constexpr auto kNEos = "get_n_eos";
 static constexpr auto kVocabSize = "get_vocab_size";
 static constexpr auto kUseKVCache = "use_kv_cache";
 static constexpr auto kUseSDPAWithKVCache = "use_sdpa_with_kv_cache";
```

Comment:
They are passed then to `std::unordered_dict<std::string, something> metadata_`. However, when you create `static constexpr auto kVocabSize = "get_vocab_size";` the deduced type will be `char*`, not std::string. To make it correct add `s` at the end, e.g. `static constexpr auto kVocabSize = "get_vocab_size"s;`